### PR TITLE
EVA-3615 - Add git commit to version and version to reports 

### PR DESCRIPTION
**Pull request by tcezard**
- Switched from using setup.py to pyproject.toml
- Added build dependency to setuptools_scm
- Use setuptools_scm to set the version
- Move executables to from the bin folder to inside the  `eva_sub_cli.executables` module to make them accessible as entry points

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    'Topic :: Scientific/Engineering :: Bio-Informatics',
+    'Topic :: Communications :: File Sharing',
+    'License :: OSI Approved :: Apache',
+]
+[project.scripts]
+'eva-sub-cli.py'='eva_sub_cli.executables.cli:main'
+'check_fasta_insdc.py'='eva_sub_cli.executables.check_fasta_insdc:main'
+'check_metadata_semantics.py'='eva_sub_cli.executables.check_metadata_semantics:main'
+'samples_checker.py'='eva_sub_cli.executables.samples_checker:main'
+'xlsx2json.py'='eva_sub_cli.executables.xlsx2json:main'
```
Comment (tcezard) [inline pyproject.toml:36]: Notes: 
- The standard format for entry points is to not have the `.py` but I've kept them for backward compatibility
- I have not found a clear easy way to `discover` the entry points

File: eva_sub_cli/executables/cli.py [hunk 2]
```diff
@@ -36,15 +36,9 @@ def validate_command_line_arguments(args, argparser):
...
 
 
-def get_version():
-    base_dir = os.path.abspath(os.path.dirname(main.__file__))
-    version = open(os.path.join(base_dir, 'VERSION')).read().strip()
-    return f'{version}'
-
-
-if __name__ == "__main__":
+def main():
```
Comment (apriltuesday) [inline eva_sub_cli/executables/cli.py:39]: This conflicts with `main` which is imported:
```
Traceback (most recent call last):
  File "/home/april/projects/eva-sub-cli/venv/bin/eva-sub-cli.py", line 8, in <module>
    sys.exit(main())
  File "/home/april/projects/eva-sub-cli/venv/lib/python3.8/site-packages/eva_sub_cli/executables/cli.py", line 80, in main
    main.orchestrate_process(**args.__dict__)
AttributeError: 'function' object has no attribute 'orchestrate_process'
```

File: README.md [hunk 3]
```diff
@@ -33,8 +33,8 @@ unzip eva-sub-cli.zip && mv eva-sub-cli-* eva-sub-cli
 Then install the library and its dependencies as follows (e.g. in a virtual environment):
 ```bash
 cd eva-sub-cli
-pip install -r requirements.txt
-python setup.py install
+# Activate your virtual environment 
+python -m pip install .
```
Comment (apriltuesday) [inline README.md:37]: I pulled your branch and tried to install, but it doesn't install the dependencies:
```
april@boudica:~/projects/eva-sub-cli$ python3 -m venv test
april@boudica:~/projects/eva-sub-cli$ source ./test/bin/activate
(test) april@boudica:~/projects/eva-sub-cli$ python -m pip install .
Processing /home/april/projects/eva-sub-cli
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Building wheels for collected packages: eva-sub-cli
  Building wheel for eva-sub-cli (PEP 517) ... done
  Created wheel for eva-sub-cli: filename=eva_sub_cli-0.4.dev36+g1592d35.d20240718-py3-none-any.whl size=268077 sha256=bcabebc0707a34a39908019958aede78488d8b023cdcf2d0d8923e6dac3ace2e
  Stored in directory: /home/april/.cache/pip/wheels/5a/f4/30/0e4664674590f68182eb9b911492e1bb33271cbe92917f5303
Successfully built eva-sub-cli
Installing collected packages: eva-sub-cli
  Attempting uninstall: eva-sub-cli
    Found existing installation: eva-sub-cli 0.4.dev36+g1592d35
    Not uninstalling eva-sub-cli at /home/april/projects/eva-sub-cli, outside environment /home/april/projects/eva-sub-cli/test
    Can't uninstall 'eva-sub-cli'. No files were found to uninstall.
Successfully installed eva-sub-cli-0.4.dev36+g1592d35.d20240718
(test) april@boudica:~/projects/eva-sub-cli$ eva-sub-cli.py --version
Traceback (most recent call last):
  File "/home/april/projects/eva-sub-cli/test/bin/eva-sub-cli.py", line 5, in <module>
    from eva_sub_cli.executables.cli import main
  File "/home/april/projects/eva-sub-cli/test/lib/python3.8/site-packages/eva_sub_cli/executables/cli.py", line 14, in <module>
    from ebi_eva_common_pyutils.logger import logging_config
ModuleNotFoundError: No module named 'ebi_eva_common_pyutils'
(test) april@boudica:~/projects/eva-sub-cli$
```
It works fine if I pip install requirements separately.

Comment (tcezard) [reply README.md:37]: Should be addressed now. Thanks for testing it.

File: eva_sub_cli/executables/cli.py (hunk 2 above)
Comment (tcezard) [reply eva_sub_cli/executables/cli.py:39]: Changed `main` -> `orchestrator`

**Review by nitin-ebi — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (apriltuesday) [reply README.md:37]: all working now :+1:

**Review by apriltuesday — APPROVED** [review]

**Pull request merged**